### ISSUE TITLE
Reference Icinga2 daemon environment in configuration

### ISSUE THREAD

**Issue by lobachpavel** [Icinga/icinga2#5904]
We are going to run icinga2 daemon in docker container and we are going to have static icinga2 configuration files for all our environments that will be a part of the docker image, but there are some config parameters which are environment specific (api user passwords, mysql password and connection settings). Our deployment process is fully automated and we are using Hashicorp Consul/Vault to store environment specific settings and credentials. The usual way we pass these environment specific parameters to dockerized applications is docker environment variables. But I don't see the way how I would reference environment variables in Icinga2 configuration. 
## Expected Behavior
Probably we need getenv() function or `env.XXXXXX` macro so we can configure it like this:
```
object ApiUser "root" {
  password = getenv("API_PASSWORD")
  permissions = [ "*" ]
}
```
```
object IdoMysqlConnection "ido-mysql" {
  user = getenv("MYSQL_USER"),
  password = getenv("MYSQL_PASSWORD"),
  host = getenv("MYSQL_HOST"),
  database = getenv("MYSQL_DB")
}
```
## Current Behavior
Currently there is no way to use Icinga2 daemon's environment variables in configuration files. And the only way is to generate configuration files dynamically using template and we want to avoid this step to keep configuration static and environment agnostic
## Possible Solution
See above
## Context
See above

**Comment by dnsmichi** [comment]
There's some issues with that request.

- if we would have sort of getenv() this is static at configuration compile time. Meaning to say, it won't be re-read on each time the attribute is referenced. 
- Dynamic access to runtime/environment attributes does not work with interfaces depending on the value, e.g. DB IDO. 
- Environment variables are typically used by plugins, and they should read from the environment upon execution. The Icinga process should not know about e.g. a MySQL root password, as this will be shared with *any* plugin being executed.
- If you need sort of ApiUser settings during container startup, solve such issues with a pre startup script. You'll have such highly likely for starting the daemon.

Last but not least, debugging such things becomes tremendously hard from the outside.

**Comment by lobachpavel** [comment]
@dnsmichi, thank you for your quick reply, see my comments below
* I'm not expecting this value will be reread at runtime, it's very unlikely icinga2 process environment will change in time :) .
* I didn't get this point
* In our case it's not a problem, since our plugins are 100% custom-written and besides in any case plugins are always under your control, icinga2 executes only what you configured.
* It means we have to recreate config files every time container starts, using some templates. The idea was to have completely static configuration (easy to debug, easy to maintain, easy to review and track change history). All our hosts and services get created via API dynamically, so apart from credentials and couple parameters we will not have anything environment specific.

For IDO we can try to configure it using API also, but we can't configure API user access via API (chicken or egg problem).

**Comment by gunnarbeutner** [comment]
I'm OK with this. :P

**Comment by dnsmichi** [comment]
Had an idea and implemented this with #6786.
